### PR TITLE
Remove LegacyBooleanAttribute

### PR DESCRIPTION
**Pull request by jpinner-lyft**

### CODE REVIEW

File: pynamodb/tests/test_attributes.py [hunk 1]
```diff
@@ -520,7 +480,7 @@ def test_boolean_deserialize(self):
         """
         attr = BooleanAttribute()
         assert attr.deserialize('1') is True
-        assert attr.deserialize('0') is True
+        assert attr.deserialize('0') is False
```
Comment (jpinner-lyft) [inline pynamodb/tests/test_attributes.py]: @garrettheel this seems like it might be a real bug in master

**Comment by jmphilli** [comment]
i'm for this - i wanted to do this back when we went from 2->3 i think

File: pynamodb/tests/test_attributes.py [hunk 2]
```diff
@@ -519,8 +479,6 @@ def test_boolean_deserialize(self):
         BooleanAttribute.deserialize
         """
         attr = BooleanAttribute()
-        assert attr.deserialize('1') is True
-        assert attr.deserialize('0') is True
         assert attr.deserialize(True) is True
```
Comment (betamoo) [inline pynamodb/tests/test_attributes.py]: should we leave the tests of of 1 and 0 too?

Comment (jpinner-lyft) [reply pynamodb/tests/test_attributes.py]: /shrug i deleted the backwards-compatibility so they won't work anyways

Comment (betamoo) [reply pynamodb/tests/test_attributes.py]: this will only be in 4.x right? So that current user should not be affect

**Comment by betamoo** [comment]
+1

File: pynamodb/tests/test_attributes.py (hunk 2 above)
Comment (jpinner-lyft) [reply pynamodb/tests/test_attributes.py]: correct -- this is for the `4.x` release so anyone who has followed along with the migration code we released and done the upgrade should not be affected

**Pull request merged**